src/lib/RoomConnection.ts

```diff
@@ -212,6 +222,18 @@ export default class RoomConnection extends TypedEventTarget {
         );
     }
 
+    private _handleClientMetadataReceived({ payload: { clientId, displayName } }: ClientMetadataReceivedEvent) {
+        const remoteParticipant = this.remoteParticipants.find((p) => p.id === clientId);
+        if (!remoteParticipant) {
+            return;
+        }
+        this.dispatchEvent(
+            new CustomEvent("participant_display_name_changed", {
```

Comment:
I guess we could in theory have some other metadata property changing here, and not the display name. For that scenario I guess we would either need to keep local participant state in this class to determine if the display name had changed, or just emit a generic "metadata updated" event which eg a reducer could handle